### ISSUE TITLE
Add update functionality to state in PendingClose and Dispute channels

### ISSUE THREAD

**Issue by marsella** [boltlabs-inc/zeekoe#166]
In some spots, a customer channel with status `PendingClose` or `Dispute` may receive information from the chain that causes them to remain in the same status, but update their internal state.

The statuses `PendingClose`, `Dispute`, and `Close` currently hold the zkAbacus type `ClosingMessage`, which contains the `CloseState` and a merchant signature on the close state. The latter is needed for mutual close operations (is it possible to not save that at all? Double-check whether it is used in the same function in which it is received).

However, zkAbacus does not reveal a function to update balances in a `CloseState`. At one point, @marsella removed a struct that held just the relevant closing information (https://github.com/boltlabs-inc/zeekoe/pull/159/files#diff-069dc1c0a24db380c354112e736c78c553a4aa8a5099cf1410c37133ed6762b8L42-L48). I think we should re-add this and also add an `update` function on the struct, which takes new balances, makes sure they add up to the same amount, and replaces with them.

Integration is blocked by #157 so should be a separate issue.
